sourcery_analytics/extractors.py

```diff
@@ -122,26 +126,30 @@ def from_function(
         """Construct an arbitrary extractor from a function of a node."""
         return cls(FunctionVisitor(function))
 
-    @functools.singledispatchmethod
     def extract(self, item: Extractable) -> typing.Iterator[T]:
         """Extract from source code, a node, a file, or directory."""
+        # `singledispatchmethod` confuses mypy, so wrap with a mypy-friendly interface
```

Comment:
**suggestion (code_clarification):** Clarify the workaround for mypy with `singledispatchmethod`.

It's good practice to provide a link to relevant mypy issues or discussions that explain why this workaround is necessary. This helps future maintainers understand the context and evaluate when the workaround might no longer be needed.

```suggestion
        # Workaround: `singledispatchmethod` confuses mypy, necessitating a wrapper for type checking.
        # This issue arises because mypy currently struggles with dynamically generated methods like those
        # created by `singledispatchmethod`. For more context and ongoing discussions, see:
        # https://github.com/python/mypy/issues/2904
        # This workaround wraps the method in a mypy-friendly interface, allowing type checking to proceed
        # without errors. Future maintainers should check the linked issue to determine if this workaround
        # remains necessary.
```